src/components/shared/Modal.js

```diff
@@ -23,18 +23,20 @@ type ModalProps = {
 export const transitionTimeout = 175;
 
 export class Modal extends React.Component<ModalProps> {
-  onClick = (e: SyntheticEvent<HTMLElement>) => {
+  onMouseDown = (e: SyntheticEvent<HTMLElement>) => {
     e.stopPropagation();
   };
 
   render() {
     const { additionalClass, children, handleClose, status } = this.props;
 
     return (
-      <div className="modal-wrapper" onClick={handleClose}>
+      <div className="modal-wrapper">
         <div
           className={classnames("modal", additionalClass, status)}
-          onClick={this.onClick}
+          onMouseDown={this.onMouseDown}
```

Comment:
Mouse down is not going to be accessible via keyboard, it should either remain onClick or have explicit handling of the " "/"Enter" key event.